flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusReporter.java

```diff
@@ -35,7 +35,7 @@
  * {@link MetricReporter} that exports {@link Metric Metrics} via Prometheus.
  */
 @PublicEvolving
```

Comment:
Add `@InstantiateViaFactory` annotation so that all instantiations go through the factory.